src/exportstep.cpp

```diff
@@ -4,6 +4,339 @@
 // Copyright 2008-2013 Jonathan Westhues.
 //-----------------------------------------------------------------------------
 #include "solvespace.h"
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+#include <functional>
+
+// Helper function to normalize coordinate values for STEP export
+// This avoids tiny floating point errors like 14.9999999993 that lead to open edges
+static double normalizeCoordinate(double val) {
```

Comment:
The rounding that this function does can be achieved much easier and the function becomes redundant.

All you need to do is round the values in the `fprintf` statements to five decimal places (search for "%." in the file). This should solve the problem since LENGTH_EPS = 1e-6 (defined [here](https://github.com/solvespace/solvespace/blob/master/src/solvespace.h#L129)) is used everywhere as a threshold for numerical solutions. Therefore all numbers (like your 14.9999999993 example) should be either ".00000" or ".99999" up to the fith decimal place and will round to the "intended" value.

For example instead of :

```
fprintf(f, "#%d=CARTESIAN_POINT('',(%.10f,%.10f,%.10f));\n"`
```

use:

```
fprintf(f, "#%d=CARTESIAN_POINT('',(%.5f,%.5f,%.5f));\n"`
```